src/Microsoft.Health.Fhir.Core/Features/Security/Authorization/SMARTScopeFhirAuthorizationService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.Core.Features.Security.Authorization
+{
+    /// <summary>
+    /// determines access based on the SMART scope.
+    /// </summary>
+    public static class SMARTScopeFhirAuthorizationService
+    {
+        public static DataActions CheckSMARTScopeAccess(RequestContextAccessor<IFhirRequestContext> requestContextAccessor, DataActions dataActions)
+        {
+            var allowedResourceActions = requestContextAccessor.RequestContext.AccessControlContext.AllowedResourceActions;
+            var resourceRequested = requestContextAccessor.RequestContext.ResourceType;
+
+            DataActions permittedDataActions = 0;
+            foreach (ScopeRestriction scopeRestriction in allowedResourceActions)
+            {
+                if (scopeRestriction.Resource == resourceRequested)
```

Comment:
I just realized one scneario which we missed.  The reource in the scopeRestriction could be '*'  for all resources.  This if statement should be updated to be == to resourceRequested or == to '*'.